### PR TITLE
Add track audit logs

### PR DESCRIPTION
**Pull request by VortexHero**

### CODE REVIEW

File: src/app/_components/sidebar.tsx [hunk 1]
```diff
@@ -154,17 +167,34 @@ export async function Sidebar({
...
-          </div>
+          {profile ? (
+            <Link
+              href={`/@${profile.username}`}
+              className="group flex items-center gap-2 overflow-hidden"
+            >
+              {/* eslint-disable-next-line @next/next/no-img-element */}
+              <img
+                src={session.user.image}
+                alt="Profile Picture"
```
Comment (dylhack) [inline src/app/_components/sidebar.tsx]: > https://html.spec.whatwg.org/multipage/images.html#a-graphical-representation-of-some-of-the-surrounding-text

`alt` should be reserved for images that hold information that the user needs to know. We can set this to an empty `alt` to mark it unimportant.

File: src/app/_components/sidebar.tsx [hunk 2]
```diff
@@ -154,17 +167,34 @@ export async function Sidebar({
...
+              <span className="max-w-[120px] truncate bg-gradient-to-br from-purple-500 to-violet-500 bg-clip-text font-semibold text-white transition group-hover:text-transparent">
+                {`@${profile.username}`}
+              </span>
+            </Link>
+          ) : (
+            <div className="flex items-center gap-2 overflow-hidden">
+              {/* eslint-disable-next-line @next/next/no-img-element */}
+              <img
+                src={session.user.image}
+                alt="Profile Picture"
```
Comment (dylhack) [inline src/app/_components/sidebar.tsx]: same thing here.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -148,6 +158,7 @@ model Track {
     musicStatus      TrackMusicStatus
     visualStatus     TrackVisualStatus
     type             TrackType           @default(ORIGINAL)
+    auditLogs        TrackAuditLog[]
```
Comment (dylhack) [inline prisma/schema.prisma:161]: what's the relationship here?

File: src/app/dashboard/projects/[project]/tracks/[track]/logs/page.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  const track = await api.track.getTrack({ username: params.track });
+
+  if (!track) {
+    return redirect(
+      `/dashboard/projects/${params.project}/tracks/${params.track}`
+    );
+  }
+
+  console.log(track);
```
Comment (dylhack) [inline src/app/dashboard/projects/[project]/tracks/[track]/logs/page.tsx]: console.log

File: prisma/schema.prisma [hunk 5]
```diff
@@ -165,6 +176,27 @@ model TrackCollaborator {
...
+    discordUsername       String?
+    discordAvatar         String?
+    targetUser            User?               @relation("TargetUser", fields: [targetUserId], references: [id])
+    targetUserId          String?
+    targetDiscordUserId   String?
+    targetDiscordUsername String?
+    targetDiscordAvatar   String?
+    action                TrackAuditLogAction
+    value                 String?
+    oldValue              String?
```
Comment (dylhack) [inline prisma/schema.prisma]: Can value be expanded to a polymorphic relationship?

Comment (dylhack) [reply prisma/schema.prisma]: or a JSON column since we're in postgres

File: src/server/api/routers/audit-log.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            return {
+              id: auditLog.id,
+              action: auditLog.action,
+              title: user ? `Track created by ${user}` : "Track created",
+              date: auditLog.createdAt,
+            };
+          }
+
+          if (auditLog.action === "UPDATE_TRACK") {
+            // TODO
```
Comment (dylhack) [inline src/server/api/routers/audit-log.ts:99]: TODO

File: prisma/schema.prisma [hunk 7]
```diff
@@ -165,6 +176,27 @@ model TrackCollaborator {
     role            TrackCollaboratorRole
 }
 
+model TrackAuditLog {
+    id                    String              @id @default(cuid())
+    track                 Track               @relation(fields: [trackId], references: [id], onDelete: Cascade)
+    trackId               String
```
Comment (dylhack) [inline prisma/schema.prisma:182]: add index on `trackId`

File: prisma/schema.prisma (hunk 3 above)
Comment (VortexHero) [reply prisma/schema.prisma:161]: wdym?

File: src/server/api/routers/audit-log.ts (hunk 6 above)
Comment (VortexHero) [reply src/server/api/routers/audit-log.ts:99]: Updating tracks hasn't been implemented yet.

File: prisma/schema.prisma (hunk 5 above)
Comment (VortexHero) [reply prisma/schema.prisma]: Made it Json

**Pull request merged**